### PR TITLE
Add update method to TreeHashMap

### PR DESCRIPTION
**Pull request by ghost**
## Overview
While the existing set method has internal logic for getting, updating, and setting the internal maps encoding the tree, there was not a client API for similar logic for the keys of the map.  This PR adds an atomic "update" method that reads a value, applies a function to it, and stores back the result.

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-3986

### CODE REVIEW

File: casper/src/main/resources/Registry.rho [hunk 1]
```diff
@@ -247,6 +265,53 @@ in {
               }
             }
           }
+        } |
+        
+        contract TreeHashMapUpdater(@map, @nybList, @n, @len, update, @suffix, ret) = {
+          // Look up the value of the node at [map, nybList.slice(0, n + 1)
```
Comment (tgrospic) [inline casper/src/main/resources/Registry.rho]: It looks like missing closing bracket.

File: casper/src/main/resources/Registry.rho [hunk 2]
```diff
@@ -247,6 +265,53 @@ in {
...
+          new valCh, restCh in {
+            nodeGet!((map, nybList.slice(0, n)), *valCh) |
+            for (@val <- valCh) {
+              if (n == len) {
+                // If we're at the end of the path, set the node to newVal.
```
Comment (tgrospic) [inline casper/src/main/resources/Registry.rho]: Comment is not related to _update_ but _set_.

**Review by tgrospic — APPROVED** [review]
Very nice, just two comments about comments.

**Review by Isaac-DeFrain — COMMENTED** [review]
lgtm

**Review by Isaac-DeFrain — APPROVED** [review]

File: casper/src/main/resources/Registry.rho (hunk 1 above)
Comment (ghost) [reply casper/src/main/resources/Registry.rho]: Done

File: casper/src/main/resources/Registry.rho (hunk 2 above)
Comment (ghost) [reply casper/src/main/resources/Registry.rho]: done

**Pull request merged**